python/servo/try_parser.py

```diff
@@ -0,0 +1,171 @@
+                         wpt_tests_to_run="_webgpu",  # run only webgpu cts
+                         profile="production",  # WebGPU works to slow with debug assert
+                         unit_tests=False)  # production profile does not work with unit-tests
+    else:
+        return None
+
+
+class Encoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, (Config, JobConfig)):
+            return o.__dict__
+        if isinstance(o, Layout):
+            return o.to_string()
+        return json.JSONEncoder.default(self, o)
+
+
+class Config(object):
+    def __init__(self, s: str | None = None):
+        self.fail_fast: bool = False
+        self.matrix: list[JobConfig] = list()
+        if s is not None:
+            self.parse(s)
+
+    def parse(self, input: str):
+        input = input.lower().strip()
+
+        if not input:
+            input = "full"
+
+        words: list[str] = input.split(" ")
+
+        for word in words:
+            # Handle keywords.
+            if word in ["fail-fast", "failfast", "fail_fast"]:
+                self.fail_fast = True
+                continue  # skip over keyword
+            if word == "full":
+                words.extend(["linux-wpt", "macos", "windows", "android"])
+                continue  # skip over keyword
+
+            preset = handle_preset(word)
+            if preset is None:
+                print(f"Ignoring unknown preset {word}")
+            else:
+                self.matrix.append(preset)
+
+    def to_json(self) -> str:
+        return json.dumps(self, cls=Encoder)
+
+
+def main():
+    conf = Config(" ".join(sys.argv[1:]))
+    print(conf.to_json())
+
+
+if __name__ == "__main__":
+    main()
+
+
+import unittest # noqa
+import logging # noqa
+
+
+class TestParser(unittest.TestCase):
+    def test_string(self):
+        self.assertEqual(Config("linux fail-fast").to_json(),
+                         '{"fail_fast": true, "matrix": [\
+{"name": "Linux", "workflow": "linux", "wpt_layout": "none", "profile": "release", \
+"unit_tests": true, "wpt_tests_to_run": ""}]}')
+
+    def test_empty(self):
+        self.assertEqual(Config("").to_json(),
+                         '{"fail_fast": false, "matrix": [\
+{"name": "Linux WPT", "workflow": "linux", "wpt_layout": "all", "profile": "release", \
+"unit_tests": true, "wpt_tests_to_run": ""}, \
+{"name": "MacOS", "workflow": "macos", "wpt_layout": "none", "profile": "release", \
+"unit_tests": true, "wpt_tests_to_run": ""}, \
+{"name": "Windows", "workflow": "windows", "wpt_layout": "none", "profile": "release", \
+"unit_tests": true, "wpt_tests_to_run": ""}, \
+{"name": "Android", "workflow": "android", "wpt_layout": "none", "profile": "release", \
+"unit_tests": false, "wpt_tests_to_run": ""}]}')
+
+    def test_full(self):
+        self.assertEqual(Config("linux-wpt macos windows android").to_json(), Config("").to_json())
```

Comment:
Right now these tests depend entirely on the order in which the keys are produced, which makes them pretty brittle. Please either sort the keys or compare them as dicts.

Reply:
I compare them as dict.